.github/workflows/ci_builds.yml

```diff
@@ -5,8 +5,10 @@ permissions:
 
 on:
   workflow_dispatch:
-  schedule:
-    - cron: '0 0,12 * * *'
+  push:
+    branches: [master, develop]
+
+concurrency: ci_build
```

Comment:
Do we know how the queue is managed? With a single lock over 2 streams, could we get a scenario where master pushes keep bumping develop builds out the queue?

Would something like `concurrency: ci-${{ github.ref }}` be slightly better, or would that saturate the build hosts?

Reply:
The reason why I chose something "generic" was to ensure it worked similarly to the develop builder we previously had -- for every new commit it "notices" it kicks off a build, otherwise it skips it.

So yes, I was trying to avoid the scenario where someone gets commit happy, does 15 merges in the space of a few minutes, and queues 15 builds while one is in progress. Given the queued workflows timeout with failure after 24 hours, and each run takes upwards of 2 hours, you could guarantee some of those later builds would fail.... and we'd most likely prefer to have the later builds complete with the intermediate ones skipped, given the state of the repo.

If we use the ref, then we only have concurrency limits placed on that ref, and all other commits will get queued. We'd need to check whether ref resolves to the branch name, or the sha1 of the commit. My only thought regarding optimisation would be to dupe this workflow for master and develop and give them their own concurrency based on branch name.

Reply:
> If we use the ref, then we only have concurrency limits placed on that ref, and all other commits will get queued. We'd need to check whether ref resolves to the branch name, or the sha1 of the commit. My only thought regarding optimisation would be to dupe this workflow for master and develop and give them their own concurrency based on branch name.

Thats what using `github.ref` would magically do

Reply:
> Thats what using `github.ref` would magically do, it would resolve to either `refs/heads/master` or `refs/heads/develop`, its not a commit ref.

In their docs they say that it resolves to different things depending on the context, which was why I suggested it needs checking.

Reply:
In the context of a push, its as ive mentioned.

Reply:
Then the concurrency would need to move to the job, given that a push to master results in a merge to develop, triggering two push events?

Reply:
Actually, splitting the workflow seems safer.

Reply:
You would still get the same amount of build events. You would have to do additional filtering to skip the commits which are a merge.

Reply:
Only the final merge should result in a push event for the auto-merge stuff, shouldn't it?
We don't get intermediate commits in say, `regen_push.yml`?

Reply:
> Actually, splitting the workflow seems safer.

Would rather we didnt do that, its worse for maintenance and given the current changes, doesnt actually offer any benefits.

Reply:
Propose an alternative?

Reply:
> Propose an alternative?
I have.

> We don't get intermediate commits in say, regen_push.yml?
regen push creates PRs on content that has already been pushed, it doesnt directly push anything.

Reply:
> regen push creates PRs on content that has already been pushed, it doesnt directly push anything.

Understood, what I was referring to was the fact that when a push occurs on develop, we don't get invocations of the `regen_push.yml` workflow for every intermediate commit, just the final merge that actually hits `develop`.

Reply:
Oh i get ya. Yeah you would just get a single push event.